### PR TITLE
support bitsandbytes quantization with more models

### PR DESCRIPTION
**Pull request by chenqianfzh**
This PR made bitsandbytes quantization available with more models. 

The model verified in my tests are:
1. Falcon
    - tiiuae/falcon-7B, quantized in flight with TP = 1 (TP =2 is not supported as the head size is 71, which is indivisible by 2)
    - prequant 4bit, alokabhishek/falcon-7b-instruct-bnb-4bit
2. opt
    - facebook/opt-125m, quantized in flight with TP = 1, 2 ,4
    - prequant 4bit: poedator/opt-125m-bnb-4bit
    - prequant 8bit: yec019/fbopt-350m-8bit
3. gemma:
    - google/gemma-7b, quantized in flight with TP = 1, 2 ,4
    - prequant 4bit: unsloth/gemma-7b-bnb-4bit
    - prequant 8bit: merve/gemma-7b-8bit
 4. gemma2:
    - google/gemma-2-9b, quantized in flight with TP = 1, 2 ,4
    - prequant 4bit: unsloth/gemma-2-9b-bnb-4bit
  5. phi
     - microsoft/phi-2, quantized in flight with TP = 1, 2 ,4
     - prequant 4bit: PrunaAI/microsoft-phi-2-pytdml-bnb-4bit-smashed
     - prequant 8bit: RichardErkhov/microsoft_-_phi-2-8bits

### CODE REVIEW

**Comment by chenqianfzh** [comment]
@mgoin @jeejeelee This is the follow-up work on bitsandbytes quantization. This PR extends the bnb support of bnb to five more models. Could you take a look?

Thanks.

File: vllm/model_executor/model_loader/loader.py [hunk 1]
```diff
@@ -736,15 +736,18 @@ def save_model(
...
-        "gate_proj", "down_proj", "up_proj", "q_proj", "k_proj", "v_proj",
-        "o_proj"
+        ".gate_proj.",
+        ".down_proj.",
+        ".up_proj.",
+        ".q_proj.",
+        ".k_proj.",
+        ".v_proj.",
+        ".o_proj.",
     ]
```
Comment (mgoin) [inline vllm/model_executor/model_loader/loader.py:749]: For future consideration, some models like [Phi3](https://huggingface.co/microsoft/Phi-3.5-mini-instruct?show_file_info=model.safetensors.index.json) have merged modules in the HF checkpoint i.e. `qkv_proj`

File: vllm/model_executor/model_loader/loader.py [hunk 2]
```diff
@@ -944,7 +948,7 @@ def _parse_quant_state(param_name: str,
         for weight_name, weight_tensor in self._hf_weight_iter(
                 hf_weights_files, use_safetensors):
             # Filter out all weights whose suffix is not ".weight"
```
Comment (mgoin) [inline vllm/model_executor/model_loader/loader.py:950]: Update or remove outdated comment

File: vllm/model_executor/models/falcon.py [hunk 3]
```diff
@@ -391,6 +391,19 @@ def forward(
...
+    default_bitsandbytes_target_modules = [
+        ".query_key_value.",
+        ".dense.",
+        ".dense_h_to_4h.",
+        ".dense_4h_to_h.",
+    ]
+
+    # BitandBytes specific attributes
+    # in TP, these weights are partitioned along the column dimension (dim=-1)
+    column_parallel_weights_modules = [".dense_4h_to_h.", ".dense."]
```
Comment (mgoin) [inline vllm/model_executor/models/falcon.py:405]: Shouldn't all of these be put under the `# BitandBytes specific attributes` comment?

**Review by mgoin — COMMENTED** [review]
Will there be some error if a user tries to enable TP for a BNB model that doesn't have column_parallel_weights_modules as an attribute?

File: vllm/model_executor/model_loader/loader.py (hunk 2 above)
Comment (chenqianfzh) [reply vllm/model_executor/model_loader/loader.py:950]: updated

File: vllm/model_executor/models/falcon.py (hunk 3 above)
Comment (chenqianfzh) [reply vllm/model_executor/models/falcon.py:405]: updated per comments.

File: vllm/model_executor/model_loader/loader.py (hunk 1 above)
Comment (chenqianfzh) [reply vllm/model_executor/model_loader/loader.py:749]: I added the known target modules to the list.

File: vllm/commit_id.py [hunk 4]
```diff
@@ -0,0 +1 @@
+__commit__ = "caf9334c9006c05b78631a6aa26daacfe9d21986"
```
Comment (mgoin) [inline vllm/commit_id.py:1]: Remove this change

File: examples/pd_sep_test.py
Comment (mgoin) [inline examples/pd_sep_test.py:1]: Remove this empty file

**Review by mgoin — APPROVED** [review]
Appreciate the small models used for testing! Just need you to remove the empty changes

File: examples/pd_sep_test.py
Comment (chenqianfzh) [reply examples/pd_sep_test.py:1]: thanks for catching this. removed.

**Comment by chenqianfzh** [comment]
@mgoin I have the PR updated. And the tests passed. Could you merge it?

Thanks.

**Pull request merged**